_posts/2024-09-24-creating-a-new-database-integration-with-data-prepper.markdown

```diff
@@ -0,0 +1,364 @@
+
+```yaml
+pipeline: 
+  workers: 2 
+  delay: 0 
+  buffer: 
+    bounded_blocking: 
+      batch_size: 125000 
+      buffer_size: 1000000 
+  source: 
+    mongodb: 
+      host: "localhost" 
+      port: 27017 
+      acknowledgments: true 
+      s3_bucket: "{{S3_BUCKET_NAME}}"  
+      s3_region: "{{S3_BUCKET_REGION}}" 
+      s3_prefix: "mongodb-opensearch"
+      insecure: "true" 
+      ssl_insecure_disable_verification: "true" 
+      authentication:
+        username: "dbuser"
+        password: "admin1234"
+      collections: 
+        - collection: "demo.demo_collection"
+          export: true
+          stream: true
+      aws:
+        sts_role_arn: "{{ROLE_ARN_FROM_STEP_5}}"
+  sink: 
+    - opensearch:  
+        hosts: [ "http://localhost:9200" ] 
+        index: "mongodb-index" 
+        document_id: "${getMetadata(\"primary_key\")}" 
+        action: "${getMetadata(\"opensearch_action\")}" 
+        document_version: "${getMetadata(\"document_version\")}" 
+        document_version_type: "external" 
+        # Default username
+        username: "admin"
+        # Change to your OpenSearch password if needed. For running OpenSearch with Docker Compose, this is set by the environment variable OPENSEARCH_INITIAL_ADMIN_PASSWORD
+        password: "OpenSearchMongoDB1#"
+        flush_timeout: -1
+```
+
+#### Step 8 - Run the pipeline
+
+With AWS credentials configured and both MongoDB and OpenSearch running on your local machine, you can launch the pipeline.
+
+First, navigate to the directory containing the Data Prepper binaries:
+
+```
+cd data-prepper/release/archives/linux/build/install/opensearch-data-prepper-$VERSION-linux-x64
+```
+
+Then, start Data Prepper using the following command:
+
+```
+bin/data-prepper
+```
+
+#### Step 9 - Review the documents in OpenSearch
+
+Wait for the export to complete, which may take a minute. Once Data Prepper displays  a log, for example, `org.opensearch.dataprepper.plugins.source.s3.ScanObjectWorker - Received all acknowledgments for folder partition`, open `http://localhost:5601` to access OpenSearch Dashboards. 
+
+Go to the **Dev Tools** application and enter `GET mongodb-index/_search` into the console editor to retrieve the MongoDB documents you created in step 2. 
+
+#### Step 10 - Add sample documents to MongoDB
+Add sample documents to MongoDB using the following command:
+ 
+```
+db.demo_collection.insertOne({"key-four": "value-four"})
+db.demo_collection.insertOne({"key-five": "value-five"})
+db.demo_collection.insertOne({"key-six": "value-six"})
+```
+
+The MongoDB source in Data Prepper will now extract  these documents from the MongoDB streams.
+
+#### Step 11 - Review the documents in OpenSearch
+
```

Comment:
Step 11 has the same title as step 9.

Reply:
Can modify these to `Step 9 - Observe the export documents in OpenSearch` and `Step 11 - Observe the stream documents in OpenSearch`

Reply:
@graytaylor0 Revised